cmd/lakefs/cmd/run.go

```diff
@@ -115,80 +114,49 @@ var runCmd = &cobra.Command{
 		dbParams := cfg.GetDatabaseParams()
 
 		var (
-			kvStore    kv.Store
-			kvParams   params.KV
 			dbPool     db.Database
 			lockDBPool db.Database
 			err        error
 		)
-		if dbParams.KVEnabled {
-			kvParams = cfg.GetKVParams()
-			kvStore, err = kv.Open(ctx, kvParams)
-			if err != nil {
-				logger.WithError(err).Fatal("Failed to open KV store")
-			}
-			defer kvStore.Close()
-
-			// Check if migration required only on postgres
-			migrationRequired := false
-			if dbParams.Type == kvpg.DriverName && len(dbParams.ConnectionString) > 0 {
-				dbPool = db.BuildDatabaseConnection(ctx, dbParams)
-				defer dbPool.Close()
-				_, _, err := db.MigrateVersion(ctx, dbPool, dbParams)
-				if err == nil {
-					migrationRequired = true
-				} else if !errors.Is(err, migrate.ErrNilVersion) {
-					logger.WithError(err).Fatal("Failed to get schema version")
-				}
-			}
-			err = kv.ValidateSchemaVersion(ctx, kvStore, migrationRequired)
-			if err != nil {
-				logger.WithError(err).Fatal("Failure on schema validation")
-			}
-		} else {
+		kvParams := cfg.GetKVParams()
+		kvStore, err := kv.Open(ctx, kvParams)
+		if err != nil {
+			logger.WithError(err).Fatal("Failed to open KV store")
+		}
+		defer kvStore.Close()
+
+		// Check if migration required only on postgres
+		migrationRequired := false
+		if dbParams.Type == kvpg.DriverName && len(dbParams.ConnectionString) > 0 {
 			dbPool = db.BuildDatabaseConnection(ctx, dbParams)
 			defer dbPool.Close()
-			lockDBPool = db.BuildDatabaseConnection(ctx, dbParams)
-			defer lockDBPool.Close()
-
-			if err := db.ValidateSchemaUpToDate(ctx, dbPool, dbParams); errors.Is(err, db.ErrSchemaNotCompatible) {
-				logger.WithError(err).Fatal("Migration version mismatch, for more information see https://docs.lakefs.io/deploying-aws/upgrade.html")
-			} else if errors.Is(err, migrate.ErrNilVersion) {
-				logger.Debug("No migration, setup required")
-			} else if err != nil {
-				logger.WithError(err).Warn("Failure on schema validation")
+			_, _, err := db.MigrateVersion(ctx, dbPool, dbParams)
```

Comment:
What if dirty == true? Do we continue?

Reply:
Just raising a question - I guess that this will be covered in the ValidateSchemaVersion check